evalml/tests/pipeline_tests/test_pipelines.py

```diff
@@ -2050,6 +2050,26 @@ class BinaryPipeline(BinaryClassificationPipeline):
     assert len(preds) == 1000
 
 
+@patch("evalml.pipelines.components.XGBoostClassifier.fit")
+def test_xgboost_ohe(mock_fit):
```

Comment:
is the interaction between OHE and xgboost a special case we need to be careful of?

Reply:
I'm wondering if we need the test as well. If the purpose is to check that Xgboost works well with bools, then we'd need to not mock fit?

Reply:
Ah fair, I can remove it!